components/ItemsList.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import {
+  ScrollView,
+  Text,
+  SectionList,
+  Button,
+  View
+} from 'react-native';
+
+import Item from './Item'
+import { styles } from './styles/stylesItemsList'
+import AddField from './AddField'
+
+export default class ItemsList extends React.Component {
+
+  constructor(props) {
+    super(props)
+    const object = props.object
+    //convert iten to sectionData
+    const sectionsData = Object.keys(object).map(function(key, index) {
+      return {
+        title: key,
+        data: Array.isArray(object[key]) ? object[key] : [object[key]],
+      }
+    });
+    this.state = {
+      sectionsData: sectionsData,
+      addField: false,
+    }
+  }
+
+  render() {
+    if (this.state.addField === true) {
+      return (
+        <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
+          <Text style={ styles.header }> Add Field </Text>
+          <AddField onComplete={(key, value) => this.addOrModifyField(key, value)} exitWithoutSave={ () => this.stopAddField() }/>
+        </View>
+      )
+    } else {
+      return (
+        <>
+        <Text style={ styles.header }> View Object </Text>
+          <SectionList
+            sections={this.state.sectionsData}
+            keyExtractor={(item, index) => item + index}
+            renderItem={({ item }) => <Item item={item} />}
+            renderSectionHeader={({ section: { title } }) => (
+              <Text style={styles.title}>  { title }  </Text>
+          )}
+        />
+        <Button title="Add/Modify field"
+          onPress={() => this.setState({addField: true})}/>
+        </>
+      )
+    }
+  }
+
+  addOrModifyField(key, value) {
+    let newSectionData = this.state.sectionsData
+    //if key do not exist - add to state
+    if (!this.state.sectionsData.map(obj => obj.title).includes(key)) {
+      newSectionData = [...this.state.sectionsData, {title: key, data: [value]}]
+    }
+    //if key exist - in case of array - add new value to array. If primitive - replace with new value
+    else {
+      const oldValue = (this.state.sectionsData.filter(obj => (obj.title === key))[0]).data
+      //array
+      if (Array.isArray(this.props.object[key])) {
```

Comment:
what happens after you change the object? is this condition still relevant?

Reply:
When adding fields you can only add a primitive one, and when modifying existing field you can only change primitive value or add to array. So in any case the only array fields are the ones in the origin object